libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -58,6 +57,8 @@ public class OAuthPrompt : Dialog
         private const string PersistedOptions = "options";
         private const string PersistedState = "state";
         private const string PersistedExpires = "expires";
+        private const string EmulatorUrl = "emulatorUrl";
+        private const string BotUrl = "botUrl";

```

Comment:
You probably could call this simply ServiceUrl = "serviceUrl".  
Emulator => Skill would have channelData.serviceUrl = emulatorUrl, and 
Skill => Emulator would have channelData.serviceUrl =skillUrl

Reply:
If we do this then there will be two 'ServiceUrl' in the activity in the emulator scenario

Activity.ServiceUrl and
Activity.ChannelData["serviceUrl"]

My take is that this would be somewhat confusing to developers and not a clear schema.